include/ofi_mem.h

```diff
@@ -226,6 +226,7 @@ struct smr_freestack {
 };
 
 #define smr_freestack_isempty(fs)	((fs)->top == SMR_FREESTACK_EMPTY)
+#define smr_freestack_isfull(fs)	((fs)->free == (fs)->size)
```

Comment:
Can you add to the commit message why this is being added?

Reply:
done

Reply:
I was asking about the addition of the isfull function.  I understand the init change.  :)

Reply:
fixed properly

Reply:
I'm trying to understand the intent with isfull.  Are you saying that if process A exits, it can break the correct behavior of process B trying to communicate with process C?

Reply:
If process A exits **WHILE** its free queue entries are still in processes B's recev queue, **THEN** process A Shared Memory Region can never be re-used, and it will turn into a zombie.

The way we clean up zombie processes is when every other PID in the file is DEAD, we shrink the file down to contain just the header (on exit... so we don't leave big files in peoples systems, and startup... startup is required in case everyone dies badly).

Reply:
I don't know what a 'free queue entry' is.  If this is the naming Alexia is commenting on, then I agree it needs to change.  "Free" queue entries are, well, free, meaning not in use...

What process placed the entries in process' B's receive queue?  Is that done *only* by process A?

Reply:
@a-szegel While we're here, also a question about cleanup - what happens if A sent all of its entries to B and then B exits without processing them? Does A get starved? How do the entries go back to A?

Reply:
yes, A will not be able to send anymore.

Reply:
1. All processes own their own "entries" for sending
2. Any process can place their entries in any other process's recv queue
 a. The recv queue is a linked list through different processes's Shared Memory Regions.
 b. All of the SMR's happen to exist in the same file (we are stateful), so we have to be careful about when we can reuse a region.

Reply:
SM2's Nemesis concept can still work with 1 SMR file per processes if we can have globally unique ID's and some way to look up late arrivers